src/S3/StreamWrapper.php

```diff
@@ -459,6 +459,8 @@ public function dir_rewinddir()
             $this->dir_opendir($this->openedPath, null);
             return true;
         });
+
+        return true;
```

Comment:
Since `boolCall` may return true or false depending on success or failure, this should be:
```
return $this->boolCall(function() {
...
}
```

Reply:
Yeah I'm agree with you but
if you see boolCall always return true cause this function always return true
And dir_opendir seems always return true too.

Same effect ? or I need to fix with 

```php
/**
     * This method is called in response to rewinddir()
     *
     * @return boolean true on success
     */
    public function dir_rewinddir()
    {
        return $this->boolCall(function() {
            $this->objectIterator = null;
            return $this->dir_opendir($this->openedPath, null);
        });
    }
```

Reply:
`boolCall` may return false if the `catch` block is hit, so please correct with the return statement as shown.

Reply:
Now fix normally